### PR TITLE
Automatically migrate unencrypted databases on request

### PR DESCRIPTION
**Pull request by ThetaSinner**
This is similar to the change I made [here](https://github.com/holochain/holochain/pull/2973).

I know that Lair databases are meant to be encrypted but the `lair_keystore` crate exposes the option to disable database encryption by not using default features, so it would be good if there was a way to migrate if you end up being forcibly switched by something that has embedded Lair and changed it's defaults. Looking at you Holochain :)

### CODE REVIEW

File: crates/lair_keystore/src/store_sqlite.rs [hunk 1]
```diff
@@ -163,18 +163,35 @@ impl SqlPool {
...
+            match create_configured_db_connection(&path, key_pragma.clone()) {
+                Ok(con) => con,
+                Err(e) => {
+                    if "true"
+                        == std::env::var("LAIR_MIGRATE_UNENCRYPTED")
+                            .unwrap_or_default()
+                            .as_str()
+                    {
+                        // Known SQLite error when the database is not encrypted, try to encrypt it and retry starting the server
+                        if let Some("file is not a database") =
```
Comment (ThetaSinner) [inline crates/lair_keystore/src/store_sqlite.rs:176]: Could improve this by changing the connection function to not return a `one_err::OneErr` and match the real type. Then convert if it doesn't match.

Comment (ThetaSinner) [reply crates/lair_keystore/src/store_sqlite.rs]: Much better

**Comment by ThetaSinner** [comment]
Waiting for a new version of sodoken https://github.com/holochain/sodoken/pull/22

**Review by neonphog — APPROVED** [review]
This looks good. It's too bad sqlcipher didn't do a vacuum into integration `VACUUM INTO bla AS ENCRYPTED KEY bla`.

**Comment by ThetaSinner** [comment]
That would be more streamlined!

I've got the build for windows working by fetching the `libsodium` lib from vcpkg instead of the `libsodium` website. That should be a more reliable source. This only fixes our CI and anybody else building on Windows might run into the same issue but I suspect it's repeated downloads that get blocked rather than individuals. I didn't see this problem locally until I rebuilt a bunch of times

**Review by neonphog — APPROVED** [review]
Nice, I'm happy with this solution. Thanks @ThetaSinner !

**Pull request merged**